scripts/code_style.py

```diff
@@ -174,22 +182,23 @@ def main() -> int:
     parser.add_argument('-f', '--fix', action='store_true',
                         help=('modify source files to fix the code style '
                               '(default: print diff, do not modify files)'))
-    # --files is almost useless: it only matters if there are no files
-    # ('code_style.py' without arguments checks all files known to Git,
-    # 'code_style.py --files' does nothing). In particular,
-    # 'code_style.py --fix --files ...' is intended as a stable ("porcelain")
-    # way to restyle a possibly empty set of files.
-    parser.add_argument('--files', action='store_true',
-                        help='only check the specified files (default with non-option arguments)')
+    parser.add_argument('--subset', action='store_true',
+                        help=('check a subset of the files known to git '
+                              '(default: check all files passed as arguments, '
+                              'known to git or not)'))
```

Comment:
Except that without `--subset`, with no files, this script processes all files known to git.

The `--subset` option has two consequences: if the list is empty then process no files (rather than all); and ignore files that aren't known to git (rather than processing them). The `--subset` behavior kind of makes sense, but the differences with the default don't really make sense.

So I think we should change the default behavior to: process the files known to git; with no option, process all the files known to git. This would mean `--subset` only causes the script to do nothing if the list of files is empty.

“Process all of these files whether known to git or not” should be a different option. But I'm not convinced we need such an option.

Reply:
I think the problem now is the same value (empty) of a single parameter (`FILE`) has two semantics: **empty file list** and **default file list** when we are processing the files know to git. That's why I suggested a dummy target to indicate the default list.

Anyway, I'm fine with your suggestions. If it also looks good to @davidhorstmann-arm, I will do the change.

Reply:
This seems fine to me. I think the semantics will always be somewhat confusing, but this seems like the least confusing option. I tend to agree that we don't need an option to check specific files unknown to git. Users who want to do this can do:
```
uncrustify -c .uncrustify.cfg --check file1.c file2.c
```